src/plugins/intel_cpu/src/nodes/interpolate.cpp

```diff
@@ -366,135 +441,1017 @@ struct jit_uni_interpolate_kernel_f32 : public jit_uni_interpolate_kernel, publi
+
+            vpermd(ymm_reg, ymm_indices, ymm_reg);
+        } else {
+            Xbyak::Zmm zmm_reg = Xbyak::Zmm(vmm_reg.getIdx());
+            Xbyak::Zmm zmm_indices = Xbyak::Zmm(vmm_indices.getIdx());
+
+            vpermd(zmm_reg, zmm_indices, zmm_reg);
+        }
+    }
+
+    inline void linear_c_gathered_1sp(int diaOW, int indexOffset, int step, bool save_weights) {
+        Xbyak::Label loop_ix_label;
+        Xbyak::Label loop_ix_end;
+        Xbyak::Label skip_loop_ix;
+
+        mov(aux_reg_weight_ow, reg_weight_ow);
+        add(reg_weight_ow, diaOW * sizeof(float));
+
+        L(loop_ix_label);
+        {
+            Xbyak::Xmm xmm_weights = Xbyak::Xmm(vmm_weights.getIdx());
+
+            cmp(aux_reg_weight_ow, reg_weight_ow);
+            jge(loop_ix_end, T_NEAR);
+
+            uni_vbroadcastss(vmm_weights, ptr[aux_reg_weight_ow]);
+            uni_vucomiss(xmm_weights, xmm_zero_const);
+            je(skip_loop_ix, T_NEAR);
+
+            mov(reg_index_ow, ptr[aux_reg_weight_ow + indexOffset * sizeof(float)]);
+            if (jcp_.spatial_dim_size > 1) {
+                uni_vmulps(vmm_weights, vmm_weights, vmm_weightsDH);
+                add(reg_index_ow, reg_index_oh);
+            }
+            uni_vaddps(vmm_weights_sum, vmm_weights_sum, vmm_weights);
+
+            int blkSize = getBlockSize(isa);
+            int stride = (jcp_.layout == InterpolateLayoutType::by_channel ? jcp_.C : blkSize) * jcp_.src_data_size;
+
+            Xbyak::Reg64 reg_index_ow_64 = reg_index_ow.cvt64();
+            imul(reg_index_ow_64, reg_index_ow_64, stride);
+
+            if (save_weights) {
+                Xbyak::Xmm xmm_tmp = Xbyak::Xmm(vmm_tmp.getIdx());
+
+                if (isa == cpu::x64::sse41) {
+                    Xbyak::Label skip_reg_save;
+                    Xbyak::Xmm xmm_weights_saved_1 = Xbyak::Xmm(vmm_weights_saved_1.getIdx());
+
+                    uni_vucomiss(xmm_weights_saved_1, xmm_invalid_weights);
+                    je(skip_reg_save, T_NEAR);
+
+                    uni_vmovdqu(vmm_index_saved_2, vmm_index_saved_1);
+                    uni_vmovdqu(vmm_weights_saved_2, vmm_weights_saved_1);
+                    uni_vmovdqu(vmm_weights_saved_1, vmm_invalid_weights);
+
+                    L(skip_reg_save);
+                }
+
+                uni_vmovd(xmm_tmp, reg_index_ow);
+                if (isa == cpu::x64::avx512_core) {
+                    mov(reg_tmp_32, 1);
+                    kmovw(k_mask, reg_tmp_32);
+
+                    vpblendmd(vmm_index_saved_1 | k_mask, vmm_index_saved_1, vmm_tmp);
+                    vblendmps(vmm_weights_saved_1 | k_mask, vmm_weights_saved_1, vmm_weights);
+                } else {
+                    if (isa == cpu::x64::avx2) {
+                        vpblendd(vmm_index_saved_1, vmm_index_saved_1, vmm_tmp, 1);
+                    } else {
+                        uni_vpblendw(vmm_index_saved_1, vmm_index_saved_1, vmm_tmp, 3);
```

Comment:
Why we need to add new uni_* mnemonic here given it is used only for SSE code path?

Reply:
Not only for SSE, but for AVX too.